tests/cspc/provisioning/provisioning_test.go

```diff
@@ -383,3 +410,140 @@ func ProvisioningTest(poolType string, bdCount int) {
+
+			})
+			// Here we are only checking for CSPI creation as passing
+			// parameters like priority class, tolerations can cause the pool
+			// manager pod to be in pending state.
+			// The intent of the tests here is to only check whether the tunables are
+			// passed or not.
+			Specify("Expected number of CSPI should be created", func() {
+				gotHealthyCSPiCount := cspcsuite.
+					client.
+					GetCSPICountEventually(cspc.Name, cspc.Namespace, cspcsuite.infra.NodeCount)
+				Expect(gotHealthyCSPiCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+			})
+
+			Specify("Expected number of pool manager deployments should be created", func() {
+				gotHealthyCSPiCount := cspcsuite.
+					client.
+					GetPoolManagerCountEventually(cspc.Name, cspc.Namespace, cspcsuite.infra.NodeCount)
+				Expect(gotHealthyCSPiCount).To(BeNumerically("==", int32(cspcsuite.infra.NodeCount)))
+			})
+
+			Specify("Resource limits should be passed to the CSPI", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasResourceLimitOnCSPIEventually(cspc.Name, cspc.Namespace, NewResourceLimit())
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Resource limits should be passed to the cstor-pool container", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasResourceLimitOnPoolManagerEventually(cspc.Name, cspc.Namespace, NewResourceLimit())
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Tolerations should be passed to the CSPI", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasTolerationsOnCSPIEventually(cspc.Name, cspc.Namespace, NewTolerations())
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Tolerations should be passed to the pool manager deployments", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasTolerationsOnPoolManagerEventually(cspc.Name, cspc.Namespace, NewTolerations())
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Priority class should be passed to the CSPI", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasPriorityClassOnCSPIEventually(cspc.Name, cspc.Namespace, &priorityClassName)
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Priority class should be passed to the pool manager deployments", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasPriorityClassOnPoolManagerEventually(cspc.Name, cspc.Namespace, &priorityClassName)
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("Compression should be passed to the CSPI", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasCompressionOnCSPIEventually(cspc.Name, cspc.Namespace, compression)
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
+			Specify("RO threshold should be passed to the CSPI", func() {
+				resoureLimitMatches := cspcsuite.
+					client.HasROThresholdOnCSPIEventually(cspc.Name, cspc.Namespace, &roThreshold)
+				Expect(resoureLimitMatches).To(BeTrue())
+			})
+
```

Comment:
Q: Shold we need to add test cases by removing reconciliation that was added and verifying whether they were reflected on the pool manager?

Reply:
Will do in the next PRs.